src/edges_analysis/analysis/beams.py

```diff
@@ -498,13 +498,15 @@ def sky_convolution_generator(
         el_above_horizon = el[horizon_mask]
 
         # Selecting sky data above the horizon
-        sky_above_horizon = sky_map[horizon_mask, :]
+        sky_above_horizon = np.zeros_like(sky_map)
+        sky_above_horizon = sky_map[horizon_mask,:]
```

Comment:
this is not going to do what you want it to. The second line will overwrite the first. You might want something like

```python
sky_above_horizon = np.nan * np.zeros_like(sky_map)
sky_above_horizon[horizon_mask] = sky_map[horizon_mask]
```

Then when we do the product of beam and sky, we have to do a nan product (if we don't already). Problematically, in terms of plotting this, the healpix will be such that the structure on the sky will remain in the same place, but the horizon cut will change place. Usually we want to plot it by having the structure move on the sky, and the horizon stay in the same place.